### PR TITLE
Fix daily bug

### PR DESCRIPTION
**Pull request by iterion**
I noticed some behavior where an extra day would be added to certain daily schedules.

This fixes those issues and adds a test case to cover.

Thanks in advance for the review.

### CODE REVIEW

**Comment by shavz** [comment]
@iterion currently going through this PR, will merge today - sorry for the delay

File: src/lib.rs [hunk 1]
```diff
@@ -1187,7 +1190,7 @@ fn lens_iter_dates(
 ) -> Vec<DateTime<Tz>> {
     let mut lensed_dates_list: Vec<DateTime<Tz>> = Vec::new();
     for date in dates_list.iter() {
-        if date.gt(&lens_from_date) {
+        if date.ge(&lens_from_date) {
```
Comment (shavz) [inline src/lib.rs]: For this function, the motivation was to strictly select future dates, as references in the comments.

For the functionality request, I might suggest adding a flag - perhaps, `incl_curr_date` which switches the comparison from `.gt` to `.ge`

Comment (iterion) [reply src/lib.rs]: Yep, I think I accidentally made this change as it actually has no impact on any of the test cases. We should probably add a test case to make sure it doesn't get changed again. 😄

Comment (shavz) [reply src/lib.rs]: agree - I'll add a test case in a seperate PR

**Pull request merged**